feat: Add MSVC build instructions (#139)
PrismLauncher/prismlauncher.org: src/wiki/development/build-instructions.md

**If this doesn't work for you, please let us know on our Discord sever, or Matrix Space.**

## Windows MSVC

### Dependencies

- [Visual Studio](https://visualstudio.microsoft.com/downloads/) - Software Distribution and Building Platform for Windows
  - If you don't want install the Visual Studio IDE, go to 'Tools For Visual Studio' and download 'Build Tools for Visual Studio' instead
  - Select 'Desktop development with C++', note that in the optional components (right side) CMake will be selected
- [Java Development Kit 8 or later](https://adoptium.net/)
  - Make sure that "Set JAVA_HOME variable" is enabled in the Adoptium installer.
- [QT](https://www.qt.io/download-qt-installer)

>   - For Qt 5.15.2, OpenSSL Toolkit is required

Reviewer: ```suggestion
  - For Qt 5 (Qt 5.15.2 is the recommended one), OpenSSL Toolkit is required
```

>   - For Qt 6.4.0, 'Qt 5 Compatibility Module' & 'Qt Image Formats' are required

Reviewer: ```suggestion
  - For Qt 6 (6.4.0 is the recommended one), 'Qt 5 Compatibility Module' & 'Qt Image Formats' are required
```

4. Now, wait for it to compile. This could take some time, so hopefully it compiles properly.
5. Run the command `cmake --install build --config Debug`, and it should install Prism Launcher to whatever the `-DCMAKE_INSTALL_PREFIX` was.
6. If you don't want Prism Launcher to store its data in `%APPDATA%`, run `cmake --install build --config Debug --component portable` after the install process
7. When building on Qt 5, whenever compiling, the OpenSSL DLLs aren't put into the directory to where Prism Launcher installs which are necessary in that case, meaning that you cannot log in. The best way to fix this, is just to do `robocopy D:/Qt/Tools/OpenSSL/Win_x64/bin/ install libcrypto-1_1-x64.dll libssl-1_1-x64.dll`. This should copy the required OpenSSL DLLs to log in. When building on Qt 6 this is not necessary because it can use schannel, the Windows tls library.

   - Replace `<Path to Qt Install>` with the path to your Qt install
   - If building for 32bit, replace `Win_x64` with `Win_x86` and remove `-x64` from the dlls names

#### Using ccache

> CMake & msbuild currently dose not support `CMAKE_CXX_COMPILER_LAUNCHER`, so the process of setting up [ccache](#ccache) support differs from other build systems

Reviewer: ```suggestion
CMake with the msbuild generation currently does not support `CMAKE_CXX_COMPILER_LAUNCHER`, so the process of setting up [ccache](#ccache) support differs from other build systems
```
